### ISSUE TITLE
how to skip Warning: Stereochemistry is wrong, using the distance geometry method instead

### ISSUE THREAD

**Issue by DreamMemory001** [openbabel/openbabel#2530]
## Environment Information
Open Babel version: 3.1.1 
Operating system and version: Linux

## Steps to Reproduce

When I gen3D for 100k smiles,  show "Open Babel Warning  in CorrectStereoAtoms“ and "Warning: Stereochemistry is wrong, using the distance geometry method instead". But the distance geometry method will spend much time in this process, I hope I can skip it and save time. I use the `eventlet`, set time out with signal, func_timeout and so on, but it not work. Thank U. I am looking forward to your reply.
![5fbb9aed53bebd3f79fe46fd17f977d](https://user-images.githubusercontent.com/32425458/187076451-916f15fb-a7eb-4afd-91a5-7311e18607c1.jpg)

**Comment by ghutchis** [comment]
You'd be willing to generate incorrect stereochemistry to save time?

**Comment by DreamMemory001** [comment]
Thank you for you reply. I hope if when it generates incorrect stereochemistry, just skip it and don`t continue use the distance geometry method instead. Because when i test in my dataset,  some smiles will spend 30 minuets or more with the distance geometry method.  

But i used the some methods with set time out with signal or thread to skip the cycle of using the distance geometry method, it not work, I guess in the distance geometry method, are you using Multi-threaded concurrency to control it? How can i skip it. Thank you very much.
![image](https://user-images.githubusercontent.com/32425458/187105228-1c737132-a1e7-451d-89b6-fed664c37850.png)

**Comment by robertwerner67** [comment]
Maybe something with the SMILES you process is wrong, which could be accidental (e.g., manual entry to a database), or systematic (a computer program generates the molecules/SMILES).  Especially if it is for the later, it would be worth to identify the SMILES currently causing the problem in first place before engaging the generation of 2D or 3D structure for a .mol file.

It is possible to run the program with SMILES as input as well as output format, e.g., by -ocan.  Since SMILES of one (more complex) structure may differ depending on the algorithm/implementation used (e.g., universal, inchified, canonical SMILES in OpenBabel), this often is a good first (or zeroth) step when working with SMILES from different sources/databases anyway.  To retain the problematic SMILES, it then might be a question of reading out the error stream, or a try-escape clause (in Python) to check if it is an error in SMILES syntax causing OpenBabel's message.

If the molecule in question contains a great number of bonds which -- in principle -- are flexible to rotation, the conversion SMILES -> 3D structure may take substantially more time, than for rigid molecules (note that --gen3d only is one of five levels to guess a conformer, see the manual) though the syntax of the SMILES is a correct one.  If you only want one conformer to start with, and not need the of lowest energy (OpenBabel equally offers a systematic conformer search), the two-step SMILES -> 2D (--gen2d) into a .mol, and subsequent 2D mol -> 3D mol (with --gen3d) sometimes is faster (this assumes you have a valid SMILES in first place).

**Comment by DreamMemory001** [comment]
@robertwerner67  Thank you for your reply. Firstly, my datasets  of smiles from the ZINC. Before I take the smiles into gen3D, canonical SMILES also the first step in my code.  Thanks for your idea of SMILES->2D, 2D->3D. I will try it at a minutes.  And I want to skip the distance geometry method. Do you know any idea about it?  Thank U.

**Comment by robertwerner67** [comment]
@DreamMemory001 No, I can't say much about the distance geometry method OpenBabel uses.  For what I want, its results fit my needs though there are other programs around (maybe [RDKit](http://rdkit.org/) is one alternative, but it depends on your project and setup).  To illustrate further the suggested approach on a one-molecule batch, on occasion, instead of a SMILES to 3d mol conversion with an intermediate 2d structure *like*

```sh
obabel -:"c1ccncc1" --gen2d -omol | obabel -imol -h --gen3d -omol
```

I run 

```sh
obabel -:"c1ccncc1" --gen2d -omol | obabel -imol --minimize --noh -omol | obabel -imol -h --gen3d -omol
```

instead.  Then, the force field optimization processes the flat 2d structure in absence of hydrogen atoms; this became possible thanks to this [PR](https://github.com/openbabel/openbabel/pull/2184) by Geoffrey Hutchison.  Eventually (last pipe), explicit hydrogens (-h) complete the 3D molecular structure (to contrast with implicit ones, flag -d).

**Comment by CADDfarmer** [comment]
> Author

Hi, Have you solved this problem? I also want to skip the distance geometry method. I know it will generate incorrect stereochemistry, so I need to skip it and remove the related files in next command. thanks sincerely!

**Comment by jianbinwu22** [comment]
> > Author
> 
> Hi, Have you solved this problem? I also want to skip the distance geometry method. I know it will generate incorrect stereochemistry, so I need to skip it and remove the related files in next command. thanks sincerely!

hello,maybe a watch command in linux could help you.
`#!/bin/bash 
time2=$(ps axu | grep 'obabel' | sed -n '1p'|sed  's/\s\s*/ /g' | cut -d' ' -f10|cut -d':' -f2 )
time1=$(ps axu | grep 'obabel' | sed -n '1p'|sed  's/\s\s*/ /g' | cut -d' ' -f10|cut -d':' -f1 )
pid=$(ps axu | grep 'obabel' | sed -n '1p'|sed  's/\s\s*/ /g' | cut -d' ' -f2)
if (( $((time2))>10||$((time1))>1));
then 
    kill -KILL $pid
fi`
save this code as watch.sh and texting: watch bash watch.sh.

**Comment by ghutchis** [comment]
This is resolved in HEAD for v3.2 - the warning is still important, but coordinate generation is improved, including faster distance geometry.
